app/src/main/java/com/teamagam/gimelgimel/app/common/logging/Log4jDiskLogger.java

```diff
@@ -0,0 +1,221 @@
+        if (mLogger != null) {
+            mLogger.info(createLogLine(VERBOSITY_USER_INTERACTION, message));
+        }
+    }
+
+    @Override
+    public void onCreate(String message) {
+        if (mLogger != null) {
+            mLogger.trace(createLifeCycleLogLine(VERBOSITY_LIFECYCLE, "onCreate", message));
+        }
+    }
+
+    @Override
+    public void onStart(String message) {
+        if (mLogger != null) {
+            mLogger.trace(createLifeCycleLogLine(VERBOSITY_LIFECYCLE, "onStart", message));
+        }
+    }
+
+    @Override
+    public void onRestart(String message) {
+        if (mLogger != null) {
+            mLogger.trace(createLifeCycleLogLine(VERBOSITY_LIFECYCLE, "onRestart", message));
+        }
+    }
+
+    @Override
+    public void onResume(String message) {
+        if (mLogger != null) {
+            mLogger.trace(createLifeCycleLogLine(VERBOSITY_LIFECYCLE, "onResume", message));
+        }
+    }
+
+    @Override
+    public void onPause(String message) {
+        if (mLogger != null) {
+            mLogger.trace(createLifeCycleLogLine(VERBOSITY_LIFECYCLE, "onPause", message));
+        }
+    }
+
+    @Override
+    public void onStop(String message) {
+        if (mLogger != null) {
+            mLogger.trace(createLifeCycleLogLine(VERBOSITY_LIFECYCLE, "onStop", message));
+        }
+    }
+
+    @Override
+    public void onDestroy(String message) {
+        if (mLogger != null) {
+            mLogger.trace(createLifeCycleLogLine(VERBOSITY_LIFECYCLE, "onDestroy", message));
+        }
+    }
+
+    @Override
+    public void onAttach(String message) {
+        if (mLogger != null) {
+            mLogger.trace(createLifeCycleLogLine(VERBOSITY_LIFECYCLE, "onAttach", message));
+        }
+    }
+
+    @Override
+    public void onCreateView(String message) {
+        if (mLogger != null) {
+            mLogger.trace(createLifeCycleLogLine(VERBOSITY_LIFECYCLE, "onCreateView", message));
+        }
+    }
+
+    @Override
+    public void onActivityCreated(String message) {
+        if (mLogger != null) {
+            mLogger.trace(createLifeCycleLogLine(VERBOSITY_LIFECYCLE, "onActivityCreated", message));
+        }
+    }
+
+    @Override
+    public void onDestroyView(String message) {
+        if (mLogger != null) {
+            mLogger.trace(createLifeCycleLogLine(VERBOSITY_LIFECYCLE, "onDestroyView", message));
+        }
+    }
+
+    @Override
+    public void onDetach(String message) {
+        if (mLogger != null) {
+            mLogger.trace(createLifeCycleLogLine(VERBOSITY_LIFECYCLE, "onDetach", message));
+        }
+    }
+
+    private void configureLogger(String directory, String fileName, long maxFileSize, int maxBackupLogFiles) {
+        LogConfigurator logConfigurator = new LogConfigurator();
+
+        logConfigurator.setFileName(directory + File.separator + fileName);
+        logConfigurator.setRootLevel(Level.ALL);
+        logConfigurator.setFilePattern("%d %-5p [%c{2}]-[%L] %m%n");
```

Comment:
Comments here about the pattern would actually be nice.
Yes, I've asked for a comment. 
It turns out they can be useful :) !fix